modules/ROOT/pages/header-removal-policy.adoc

```diff
@@ -0,0 +1,62 @@
+= Header Removal
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+[width="100%", cols="5,15"]
+|===
+>s| Policy Name | Header Removal
+>s|Summary      | Removes headers from a request or a response 
+>s|Category | Transformation
+>s| First Mule version available | 4.1.0
+.1+>.^s| Returned Status Codes
+
+[%autowidth.spread,cols="a,a"]
+!===
+.1+>.^! 500 ! Configuration includes an expression that was not evaluated.
+! The provided expression in the configuration was not validated. +
+!===
+
+|===
+
+The Header Removal policy removes all listed headers from a request or a response of a message. When you configure this policy for your API, you must specify an inbound and outbound map of the headers that you want to remove from the message processing in the form of a key-value pair.
+
+image::header-inject-remove-diagram.png[]
+
+
+You can optionally include DataWeave expressions for Mule 4.0 and Mule runtime engine (Mule) expressions for Mule 3.0 and in the value of the header:
+
+`“#[attributes.requestPath]”`
+
+In this example, all the headers matching the configured expression are removed from the message.
+
+
+== How This Policy Works
+
+Every header matching an expression that you specify in the policy parameter is removed from a message. Therefore, you must double-check your expression for the header removal.
+
+For example, the `content-type` and `content-length` headers and the host header of a request can be filtered by this policy. In the response, only the `content-type` header is removed:
```

Comment:
```suggestion
Because every header that matches a Header Removal policy parameter is removed from the message, you must be careful about how you configure that parameter. 

For example, the following parameter configuration removes only the `content-type` header from a message that also contains `content-length` and host headers:
```

Reply:
Done! Thanks!